src/handlers/gestures/useGesture.ts

```diff
@@ -0,0 +1,215 @@
+import React from 'react';
+import RNGestureHandlerModule from '../../RNGestureHandlerModule';
+import {
+  BaseGesture,
+  GestureRef,
+  GestureType,
+  HandlerCallbacks,
+  InteractionBuilder,
+} from './gesture';
+import { registerHandler, unregisterHandler } from '../handlersRegistry';
+import Animated from 'react-native-reanimated';
+import {
+  baseGestureHandlerWithMonitorProps,
+  filterConfig,
+} from '../gestureHandlerCommon';
+import { tapGestureHandlerProps } from '../TapGestureHandler';
+import {
+  panGestureHandlerProps,
+  panGestureHandlerCustomNativeProps,
+} from '../PanGestureHandler';
+import { longPressGestureHandlerProps } from '../LongPressGestureHandler';
+import { forceTouchGestureHandlerProps } from '../ForceTouchGestureHandler';
+import { flingGestureHandlerProps } from '../FlingGestureHandler';
+
+const ALLOWED_PROPS = [
+  ...baseGestureHandlerWithMonitorProps,
+  ...tapGestureHandlerProps,
+  ...panGestureHandlerProps,
+  ...panGestureHandlerCustomNativeProps,
+  ...longPressGestureHandlerProps,
+  ...forceTouchGestureHandlerProps,
+  ...flingGestureHandlerProps,
+];
+
+export type GestureConfigReference = {
+  config: GestureType[];
+  callback: null | (() => void);
+  animatedEventHandler: any;
+  animatedHandlers: Animated.SharedValue<
+    HandlerCallbacks<Record<string, unknown>>[] | null
+  > | null;
+  firstExecution: boolean;
+};
+
+export function useGesture(gestureConfig: InteractionBuilder | GestureType) {
+  const gesture = gestureConfig.configure();
+
+  const result = React.useRef<GestureConfigReference>({
+    config: gesture,
+    callback: null,
+    animatedEventHandler: null,
+    animatedHandlers: null,
+    firstExecution: true,
+  });
```

Comment:
You can add `.current` here to decrease verbosity.

Reply:
Done in 991221b8434e0777733a763c092c962fe2d368e9